File: 卡码/代码随想录/Go/贪心/03-困难/LeetCode_0056.go
// 56. 合并区间
package main

import (
	"math"
	"sort"
)

func merge(intervals [][]int) [][]int {
	res:=make([][]int,0)
	sort.Slice(intervals,func (i,j int) bool{
		return intervals[i][0]<intervals[j][0]
	})
	res=append(res,intervals[0])
	for i:=1;i<len(intervals);i++ {
		if intervals[i][0]<=res[len(res)-1][1] {
			res[len(res)-1][1]=int(math.Max(float64(intervals[i][1]),float64(res[len(res)-1][1])))
		}else{
			res=append(res,intervals[i])
		}
	}
	return res
}

File: 卡码/代码随想录/JS/贪心/03-困难/LeetCode_0056.js
//56. 合并区间
const merge = function(intervals) {
	const res=new Array();
	intervals.sort((a,b)=>a[0]-b[0]);
	res.push(intervals[0]);
	for(let i=1;i<intervals.length;i++){
		if(intervals[i][0]<=res[res.length-1][1]){
			res[res.length-1][1]=Math.max(intervals[i][1],res[res.length-1][1]);
		}else{
			res.push(intervals[i]);
		}
	}
	return res;
};